src/expr/src/relation/mod.rs

```diff
@@ -788,8 +794,6 @@ impl MirRelationExpr {
                         key_set.sort();
                         key_set.dedup();
                     }
-                    input.sort();
-                    input.dedup();
```

Comment:
I didn't follow the reasoning for why this is not required, but if not it would be great to leave a comment with the reasoning. For example, why is it in sorted order already if for example we were to replace the first element of a key? Or perhaps we don't need sorted order (probably).

Reply:
I left [a comment here](https://github.com/MaterializeInc/materialize/pull/25708#discussion_r1528775168) but it seems that GitHub is hiding it now because I also removed the two lines in the proposed `fixup!` commit. The content was:

> nit: this and the input.dedup() can be omitted as those are also done on the result-bound keys.

This should also explain [my other comment above](https://github.com/MaterializeInc/materialize/pull/25708#discussion_r1528777440)

I mean that we can since we have

```rust
keys.sort();
keys.dedup();
```

at the end of `keys_with_input_keys`, we can possibly also do the 

```rust
for key_set in keys.iter_mut() {
    key_set.sort();
    key_set.dedup();
}
```

there and ensure that `key_set`s emitted from all cases in the 
```rust
let mut keys = match self {
    ...
}
```
branch appear in a sorted and deduped form.

Reply:
Beyond the scope of this PR, but: maybe we can introduce a `Key` struct that keeps its internal state in this canonical form (sorted, duplicate free). Then checking whether `k1: Key ⊆ k2: Key` should be doable in linear time.